rfcs/src/intro.md

```diff
@@ -0,0 +1,56 @@
+# Introduction
+
+Kani is an open-source verification tool that uses automated reasoning to analyze Rust programs. In order to 
+integrate feedback from developers and users on future changes to Kani, we decided to follow a light-weight 
+"RFC" (request for comments) process.
+
+## When to create an RFC
+
+You should create an RFC in one of two cases:
+
+1. The change you are proposing would be a "one way door": e.g. a change to the public API, a new feature that would be difficult to modify once released, deprecating a feature, etc.
+2. The change you are making has a significant design component, and would benefit from a design review.
+
+Bugs and smaller improvements to existing features do not require an RFC.
+If you are in doubt, feel free to create  a [feature request](https://github.com/model-checking/kani/issues/new?assignees=&labels=&template=feature_request.md) and discuss the next steps in the new issue.
+Your PR reviewer may also request an RFC if your change appears to fall into category 1 or 2.
+
+You do not necessarily need to create an RFC immediately. It is our experience that it is often best to write some "proof of concept" code to test out possible ideas before writing the formal RFC.```
+
+## The RFC process
+
+This is the overall workflow for the RFC process:
+
+```toml
+    Create RFC ──> Receive Feedback  ──> Accepted?
+                                            │ Y
+                                            ├───> Implement ───> Stabilize?
+                                            │ N                      │ Y
+                                            └───> Close PR           ├───> RFC Stable
+                                                                     │ N
+                                                                     └───> Remove feature
+```
+
+1. Create an RFC
+   1. Start from a fork of the Kani repository.
+   2. Copy the template file ([`rfcs/src/template.md`](./template.md)) to `rfcs/src/<my-feature>.md`.
+   3. Fill in the details according to the template intructions.
+   4. Submit a pull request.
+2. Build consensus and integrate feedback.
+   1. RFCs should get approved by at least 2 members of the `kani-developers` team.
+   2. Once the RFC has been approved, update the RFC status and merge the PR.
+   3. If the RFC is not approved, close the PR without merging.
+3. Feature implementation.
+   1. Start implementing the new feature in your fork.
+   2. It is OK to implement it incrementally over multiple PRs. Just ensure that every pull request has a testable 
+      end-to-end flow and that it is properly tested.
+   3. In the implementation stage, the feature should only be accessible if the user explicitly passes 
+      `--enable-unstable` as an argument to Kani.
+   4. Document how to use the feature.
+4. Stabilization.
+   1. After the feature has been implemented, start the stabilization process.
+   2. Gather user feedback and make necessary adjustments.
```

Comment:
We should define some criteria to stabilize a feature. This reads like "Stabilize right away after implementing it", when in practice we will wait for a feature to be stabilized after a period. Recent bug reports or short periods after implementation should all delay the stabilization of a feature.

Reply:
Yeah, I totally agree. Any suggestions?

I would prefer keeping it lightweight for now. Something like creating a PR that marks the RFC as stable and remove the `--enable-unstable` requirement. In the description of the PR, the developer states why they believe the feature can be marked as stable and we go from there. I guess we could add a minimum "bake" time like 2 releases.

 I think for the rust project, they have two different process which is more heavy weight. This includes writing a report and starting an FCP (final comment period).

Reply:
I added a stage between "Implementation" and "Stabilization". Let me know what you think.

Reply:
Thanks for adding another stage. I think that makes clear there is not an immediate path to stabilization, while at the same time keeping the process lightweight as we want.